source/pages/Receive/Receive.tsx

```diff
@@ -25,21 +26,28 @@ export const Receive = () => {
   const [loaded, setLoaded] = useState<boolean>(false);
 
   useEffect(() => {
+    const abortController = new AbortController();
+
     const setNewAddress = async () => {
-      if (
-        (activeNetwork.chainId === 57 || activeNetwork.chainId === 5700) &&
-        (await controller.wallet.account.sys.setAddress())
-      ) {
+      if (activeNetwork.chainId === 57 || activeNetwork.chainId === 5700) {
```

Comment:
This verification can lead to potential bugs, these chainId are the same for sysoin UTXO and NEVM mainnet and testnet. 
Should have an extra validation to guarantee we're just calling this function on the UTXO case. (Probably using isBitcoinBasedChain would be enough)